library/model/book.py

```diff
@@ -31,62 +26,33 @@ def __init__(self, title, authors, publisher, pub_date, genres, pages, isbn, typ
         self.genres = genres
         self.pages = pages
         self.isbn = isbn
-        self._book_type = type
-        self.duration = duration
-        self.existing_items = existing_items
-        self.borrowed_items = borrowed_items
 
     @classmethod
     def from_borrowed_book(cls, borrowed_book: "BorrowedBook") -> "Book":
         book = Book(
-            borrowed_book.title,
-            borrowed_book.authors,
-            borrowed_book.publisher,
-            borrowed_book.publication_date,
-            borrowed_book.genres,
-            borrowed_book.pages,
-            borrowed_book.isbn,
-            borrowed_book._book_type,
-            borrowed_book.duration,
-            borrowed_book.existing_items,
-            borrowed_book.borrowed_items,
+            borrowed_book.book.title,
+            borrowed_book.book.authors,
+            borrowed_book.book.publisher,
+            borrowed_book.book.publication_date,
+            borrowed_book.book.genres,
+            borrowed_book.book.isbn
         )
         return book
 
+    @abstractmethod
     def can_borrow(self) -> bool:
-        if self._book_type == "Paper":
-            return self.existing_items - self.borrowed_items > 0
-        elif self._book_type == "Electronic":
-            return True
-        elif self._book_type == "Audio":
-            return True
-        else:
-            raise AttributeError("No such book type...")
+        raise NotImplementedError("Unspecified book type")
```

Comment:
I think there's no need to raise here - the method cannot be used anyways as long as it is abstracted and must be reimplemented later on.